### PR TITLE
Add input validation and error handling for contact and kouden deletion

### PR DESCRIPTION
**Pull request by otomatty**
## Summary
This PR adds comprehensive input validation and improves error handling across the contact form and kouden deletion features. It introduces Zod schemas for validation, structured error responses, and better user feedback through toast notifications.

### Contact Request Validation
- Created `src/schemas/contact.ts` with `contactRequestSchema` for validating contact form inputs
  - Validates category enum, email format, message length (max 5000 chars), name/subject/company_name (max 100-200 chars)
  - Handles empty strings as nullable values using Zod preprocessing
  - Comprehensive test coverage in `src/schemas/__tests__/contact.test.ts`
- Updated `src/app/_actions/contact.ts` to use schema validation
  - Changed return type to `CreateContactRequestResult` (success/error tuple pattern)
  - Returns validation errors to the caller instead of throwing exceptions
  - Logs validation failures with field-level details

### Kouden Deletion Error Handling
- Updated `src/app/_actions/koudens/delete.ts` to return structured results
  - Changed return type to `DeleteKoudenResult` (success/error tuple pattern)
  - Returns specific error messages for permission denied and database errors
  - Catches unexpected exceptions and returns error instead of propagating
  - Added comprehensive test coverage in `src/app/_actions/__tests__/koudens-delete.test.ts`
- Updated `src/app/(protected)/koudens/[id]/_components/actions/delete-kouden-dialog.tsx` to handle the new result type
  - Shows error toast with specific error message on failure
  - Improved error logging

### API Pagination Validation
- Created `src/lib/api/pagination.ts` with `paginationQuerySchema` and `parsePagination` helper
  - Validates `page` (min 1) and `pageSize` (1-100) parameters
  - Prevents DoS attacks via excessively large pageSize values
  - Comprehensive test coverage in `src/lib/api/__tests__/pagination.test.ts`
- Updated `src/app/api/koudens/[id]/entries/route.ts` to use pagination validation
  - Returns 400 with validation details on invalid parameters

### UI Improvements
- Updated `src/components/contact/contact-form.tsx` and `src/components/contact/app-contact-form.tsx`
  - Added error handling for contact request failures
  - Shows toast notifications with specific error messages
  - Improved try-catch error handling

## Implementation Details
- All error-returning functions follow the `{ success: boolean; error?: string }` pattern for consistent error handling
- Validation schemas use Zod for type-safe runtime validation
- Empty form fields are converted to `null` rather than empty strings for database consistency
- All new code includes comprehensive unit tests with edge cases
- Imports organized alphabetically for consistency

https://claude.ai/code/session_012WqxjSJ7UKdk6cngMc6Qxp

## リリースノート

* **New Features**
  * ページネーション検証ユーティリティと型を追加しました。
  * お問い合わせの入力スキーマと結果型を公開しました。

* **Bug Fixes**
  * お問い合わせ送信で入力検証と失敗時の明確なエラーハンドリングを導入しました（UIでのエラー表示・再送指示）。
  * 削除処理の結果判定を厳密化し、失敗時に適切な通知と処理中断を行うようにしました。
  * フォームの未保存変更保護を簡素化（beforeunload）しました。

* **Tests**
  * 削除処理・お問い合わせ検証・ページネーションの自動テストを追加しました。

[Review Change Stack](https://app.coderabbit.ai/change-stack/otomatty/kouden/pull/61)

### CODE REVIEW

File: src/components/contact/contact-form.tsx [hunk 1]
```diff
@@ -123,8 +124,19 @@ export default function ContactForm() {
 				formData.append(key, String(value));
 			}
 		}
-		await createContactRequest(formData);
-		router.push("/contact/success");
+		try {
+			const result = await createContactRequest(formData);
```
[the remark this answers is not in the document]
Comment (otomatty) [reply src/components/contact/contact-form.tsx:114]: Out of scope for this PR (Issue #35 はバリデーション不足の修正)。添付ファイル機能の漏れは元コードから存在する既存バグで、`createContactRequest` を ID 返却型に変えて `uploadContactAttachment` を呼ぶ修正は別 Issue / PR で扱います。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/components/contact/app-contact-form.tsx [hunk 2]
```diff
@@ -99,7 +99,12 @@ export default function AppContactForm({ user, onSuccess }: AppContactFormProps)
 				}
 			}
 
-			await createContactRequest(formData);
+			const result = await createContactRequest(formData);
```
[the remark this answers is not in the document]
Comment (otomatty) [reply src/components/contact/app-contact-form.tsx:102]: Same as the contact-form thread — 既存バグで、Issue #35 のスコープ外のため別途対応とします。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/lib/api/__tests__/pagination.test.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
+/// <reference types="vitest" />
```
[the remark this answers is not in the document]
Comment (otomatty) [reply src/lib/api/__tests__/pagination.test.ts:1]: 本リポジトリの既存テストはすべて `__tests__/` サブフォルダに集約されています (`src/app/_actions/__tests__/`, `src/schemas/__tests__/`, `src/components/survey/__tests__/` 等)。新規ファイルだけ co-locate にすると一貫性が崩れるため、既存パターンに揃えています。配置ルール自体を変更する場合は別途リポジトリ全体での議論として扱うのが適切です。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/_actions/__tests__/koudens-delete.test.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		(canDeleteKouden as any).mockResolvedValue(true);
+		supabaseMock.eq.mockResolvedValue({ error: null });
+
+		const result = await deleteKouden("kouden-1");
+
+		expect(result.success).toBe(true);
+		expect(supabaseMock.from).toHaveBeenCalledWith("koudens");
+		expect(supabaseMock.delete).toHaveBeenCalled();
+		expect(supabaseMock.eq).toHaveBeenCalledWith("id", "kouden-1");
+	});
```
[the remark this answers is not in the document]
Comment (otomatty) [reply src/app/_actions/__tests__/koudens-delete.test.ts:70]: 既存テスト (`src/app/_actions/__tests__/user-surveys.test.ts` 等) でも 1 `it` 内に複数 `expect` を置くパターンが採用されており (13 `it` / 29 `expect` ≒ 2.2 expect/test)、ここだけ厳密分割するとリポジトリ全体の規約と不整合になります。本 PR では既存パターンに揃えて維持します。テスト分割ルールを徹底する場合は、別途リポジトリ全体での議論で扱うのが適切です。

---
_Generated by [Claude Code](https://claude.ai/code)_
